### DISCUSSION TITLE
Trigger automation when radio advert plays

### DISCUSSION THREAD

**Discussion by mikejclark** [music-assistant/support#1614] [Feature requests and ideas]
I use music assistant primarily for listening to radio stations on chromecast devices. I noticed that music assistant displays when an advert is playing - see image. Is there a way to use this attribute for an automation? i.e. to mute the speaker

![Screenshot_20231029_220203_Home Assistant](https://github.com/music-assistant/hass-music-assistant/assets/6953022/c06cb744-36a4-4ae0-8534-cd46225277d7)

**Comment by OzGav** [comment]
Should be able to. Look at the entity attributes when you see that as the track name and then make an automation that mutes and unmutes depending on whether a word or phrase is included. Not sure if all stations send this info though.

https://community.home-assistant.io/t/using-attribute-as-trigger-for-automation/124665/10

**Reply by mikejclark** [reply]
I have attempted this but cannot see the attribute:

![image](https://github.com/music-assistant/hass-music-assistant/assets/6953022/c047d4b8-3ac8-4c63-ad21-e9d47e4060f6)

But it does seem the music assistant pulls this from somewhere

**Reply by OzGav** [reply]
Is there a HA player as well? Does that show it?

**Reply by mikejclark** [reply]
Not that I can find. I've searched the attributes for the radio station name but only the one media player comes up

**Reply by OzGav** [reply]
Yeah I just had a play myself and “BBC world Service” shows “BBC World Service Online” which I don’t see in the attributes…. @marcelveldt ?

**Reply by mikejclark** [reply]
@marcelveldt any idea how I can achieve this? Thanks

**Reply by OzGav** [reply]
Have a look at this sgain as there is an attempt now to put ADVERT in the attributes when it is detected

**Reply by OzGav** [reply]
Duplicate of https://github.com/orgs/music-assistant/discussions/1935

**Reply by mikejclark** [reply]
Looking good so far. Was not expecting this to be implemented so I'm very pleased!

![Screenshot_20240510_103819_Home Assistant](https://github.com/music-assistant/hass-music-assistant/assets/6953022/f499b864-69fe-40e4-bd18-2f5174de92ad)

**Reply by OzGav** [reply]
Let me know how you go. I have one station that seems to be permanently broadcasting advert in their metadata so I have it on my list to look more closely at what is going on there

**Reply by mikejclark** [reply]
Seems to be working for Absolute 80s - 
stream_title changed from Advert to null. This is with a song playing: 
![Screenshot_20240510_104657_Home Assistant](https://github.com/music-assistant/hass-music-assistant/assets/6953022/4147405d-0d0e-4312-837e-a37744ba426c)
